### PR TITLE
Added Laboratory results

### PR DESCRIPTION
**Pull request by FuriousLlama**
## Status
**Ready**

## Fixes or Implements [AB#8332](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8332) & [AB#8349](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8349)
- [X] Enhancement
- [ ] Bug
- [ ] Documentation

## Description:
Added laboratory reports to the timeline.
Added laboratory vuex store
Enhanced laboratory retrieval services

## Testing
- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

### UI Changes
YES

### Browsers Tested
- [X] Chrome
- [ ] Safari
- [ ] Edge
- [ ] Firefox

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/app/components/timeline/laboratory.vue [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            <div class="d-flex flex-row-reverse">
+              <b-btn
+                v-b-toggle="'entryDetails-' + index + '-' + datekey"
+                variant="link"
+                class="detailsButton"
+                @click="toggleDetails()"
+              >
+                <span class="when-opened">
+                  <font-awesome-icon
+                    icon="chevron-down"
```
Comment (sam-warren) [inline Apps/WebClient/src/ClientApp/app/components/timeline/laboratory.vue:86]: We want chevron-up when opened and chevron-down when closed

File: Apps/WebClient/src/ClientApp/app/components/timeline/laboratory.vue [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                    aria-hidden="true"
+                  ></font-awesome-icon
+                ></span>
+                <span class="when-closed">
+                  <font-awesome-icon
+                    icon="chevron-up"
```
Comment (sam-warren) [inline Apps/WebClient/src/ClientApp/app/components/timeline/laboratory.vue:92]: chevron-down

File: Apps/WebClient/src/ClientApp/app/components/timeline/laboratory.vue [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+  private get entryIcon(): IconDefinition {
+    return faFlask;
+  }
+
+  private toggleDetails(): void {
+    this.detailsVisible = !this.detailsVisible;
+    this.hasErrors = false;
+
+    if (!this.detailsVisible) {
```
Comment (sam-warren) [inline Apps/WebClient/src/ClientApp/app/components/timeline/laboratory.vue:194]: Is this necessary?

File: Apps/WebClient/src/ClientApp/app/components/timeline/laboratory.vue (hunk 1 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/app/components/timeline/laboratory.vue:86]: I think that is how medication works now. Thats where I got it from

File: Apps/WebClient/src/ClientApp/app/components/timeline/laboratory.vue (hunk 3 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/app/components/timeline/laboratory.vue:194]: not at all! good catch

**Pull request merged**